usr/lib/zypp/plugins/urlresolver/susecloud

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+
+# Copyright (c) 2019, SUSE LLC, All rights reserved.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+"""URL Resolver zypper plugin. This plugin resolves the url to the proper
+   update server and sends along the necessary instance data."""
+
+
+import base64
+import logging
+
+import cloudregister.registerutils as utils
+
+from zypp_plugin import Plugin
+
+utils.start_logging()
+utils.set_proxy()
+
+
+class SUSECloudPlugin(Plugin):
+    def RESOLVEURL(self, headers, body):
```

Comment:
Consider snakecase `resolve_url`.

Reply:
Zypper resolver plugin entry point has to be named `RESOLVEURL`.